template/src/screens/AuthComponent/SignUpScreen.tsx

```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ScreenContainer from '../../components/ScreenContainer'
+import {AuthStackParamList} from '../../navigation/types'
+import {StackNavigationProp} from '@react-navigation/stack'
+import {RouteProp} from '@react-navigation/native'
+
+interface IProps {
+  navigation: StackNavigationProp<AuthStackParamList, 'SignUpScreen'>
+  route: RouteProp<AuthStackParamList, 'SignUpScreen'>
```

Comment:
use `RouteKey` constants

Reply:
RouteName of StackNavigationProp only accept key of AuthStackParamList

Reply:
Then we can use
```
interface IProps {
  navigation: StackNavigationProp<AuthStackParamList, keyof AuthStackParamList>
  route: RouteProp<AuthStackParamList, keyof AuthStackParamList>
}
```

Reply:
outdated